Library/OcBootManagementLib/DefaultEntryChoice.c

```diff
@@ -850,6 +844,65 @@ OcSetDefaultBootEntry (
   return Status;
 }
 
+EFI_LOAD_OPTION *
+InternalGetBoostrapOptionData (
+  OUT UINTN                    *LoadOptionSize,
+  OUT UINT16                   *BootOption,
+  OUT EFI_DEVICE_PATH_PROTOCOL **LoadPath,
+  IN  UINT16                   *BootOptions,
+  IN  UINTN                    NumBootOptions
+  )
+{
+  UINTN                    BootIndex;
+  EFI_LOAD_OPTION          *CurrLoadOption;
+  EFI_DEVICE_PATH_PROTOCOL *CurrDevicePath;
+  BOOLEAN                  IsBooptstrap;
+  //
+  // Check all boot options for trailing "\Bootstrap\Bootstrap.efi".
+  //
+  for (BootIndex = 0; BootIndex < NumBootOptions; ++BootIndex) {
+    CurrLoadOption = InternalGetBootOptionData (
+      LoadOptionSize,
+      BootOptions[BootIndex],
+      &gEfiGlobalVariableGuid
+      );
+    if (CurrLoadOption == NULL) {
+      continue;
+    }
+
+    CurrDevicePath = InternalGetBootOptionPath (
+      CurrLoadOption,
+      *LoadOptionSize
+      );
+    if (CurrLoadOption == NULL) {
+      FreePool (CurrLoadOption);
+      continue;
+    }
+
+    IsBooptstrap = OcDevicePathHasFilePathSuffix (
+      CurrDevicePath,
+      OPEN_CORE_BOOTSTRAP_PATH,
+      L_STR_SIZE (OPEN_CORE_BOOTSTRAP_PATH)
+      );
+    if (IsBooptstrap) {
+      break;
+    }
+
+    FreePool (CurrLoadOption);
+  }
+
+  if (BootIndex == NumBootOptions) {
+    return NULL;
+  }
+
+  if (LoadPath != NULL) {
+    *LoadPath   = CurrDevicePath;
+  }
+
+  *BootOption = BootIndex;
```

Comment:
Rename to BootOptionIndex. Too confusing otherwise.

Reply:
OK